### PR TITLE
improve the performance of composing client metrics in liConsumer

### PR DESCRIPTION
**Pull request by radai-rosenblatt**
Signed-off-by: Radai Rosenblatt <rrosenbl@rrosenbl-ld1.linkedin.biz>

### CODE REVIEW

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/CompositeIterator.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  public boolean hasNext() {
+    return a.hasNext() || b.hasNext();
+  }
+
+  @Override
+  public T next() {
+    if (a.hasNext()) {
+      return a.next();
+    }
+    return b.next();
```
Comment (xiowu0) [inline li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/CompositeIterator.java]: either a or b can be null

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/CompositeCollection.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    if (a == null || b == null) {
+      throw new IllegalArgumentException("arguments must not be null");
+    }
+    this.a = a;
+    this.b = b;
+  }
+
+  @Override
+  public int size() {
+    return a.size() + b.size();
```
Comment (xiowu0) [inline li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/CompositeCollection.java]: check null for a and b? Here and in other places.

Comment (radai-rosenblatt) [reply li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/CompositeCollection.java]: i do, at the constructor

File: li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/CompositeIterator.java (hunk 1 above)
Comment (radai-rosenblatt) [reply li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/CompositeIterator.java]: i check, at the constructor

**Pull request merged**